pkg/river/types.go

```diff
@@ -0,0 +1,75 @@
+package river
```

Comment:
Some of the River interfaces are hidden in internal packages, so I'm using pkg/river to document and re-expose them.